src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/ListWithPhysicalAndLogicalLengthFW.java

```diff
@@ -0,0 +1,324 @@
+            field0() != null ? field0RO.asString() : null, field1(), field2() != null ? field2RO.asString() : null);
+    }
+
+    public static final class Builder extends Flyweight.Builder<ListWithPhysicalAndLogicalLengthFW>
+    {
+        private final StringFW.Builder field0RW = new StringFW.Builder();
+
+        private final StringFW.Builder field2RW = new StringFW.Builder();
+
+        private int fieldsMask;
+
+        public Builder()
+        {
+            super(new ListWithPhysicalAndLogicalLengthFW());
+        }
+
+        private StringFW.Builder field0()
+        {
+            int newBit = fieldsMask | (1 << FIELD_INDEX_FIELD0);
+            assert newBit == (newBit & 0x01) : "Value out of order for field \"field0\" in the list";
+            return field0RW.wrap(buffer(), limit(), maxLimit());
+        }
+
+        public Builder field0(
+            String value)
+        {
+            StringFW.Builder field0RW = field0();
+            field0RW.set(value, StandardCharsets.UTF_8);
+            fieldsMask |= 1 << FIELD_INDEX_FIELD0;
+            limit(field0RW.build().limit());
+            return this;
+        }
+
+        public Builder field0(
+            StringFW value)
+        {
+            StringFW.Builder field0RW = field0();
+            field0RW.set(value);
+            fieldsMask |= 1 << FIELD_INDEX_FIELD0;
+            limit(field0RW.build().limit());
+            return this;
+        }
+
+        public Builder field0(
+            DirectBuffer buffer,
+            int offset,
+            int length)
+        {
+            StringFW.Builder field0RW = field0();
+            field0RW.set(buffer, offset, length);
+            fieldsMask |= 1 << FIELD_INDEX_FIELD0;
+            limit(field0RW.build().limit());
+            return this;
+        }
+
+        public Builder field1(
+            long value)
+        {
+            if (value < 0)
+            {
+                throw new IllegalArgumentException(String.format("Value %d too low for field \"field1\"", value));
+            }
+            assert (value & 0xffff_ffff_0000_0000L) == 0L : "Value out of range for field \"field1\"";
+            int newBit = fieldsMask | (1 << FIELD_INDEX_FIELD1);
+            assert newBit == (newBit & 0x03) : "Value out of order for field \"field1\" in the list";
+            int newLimit = limit() + FIELD_SIZE_FIELD1;
+            checkLimit(newLimit, maxLimit());
+            buffer().putInt(limit(), (int) (value & 0xFFFF_FFFFL));
+            fieldsMask |= 1 << FIELD_INDEX_FIELD1;
+            limit(newLimit);
+            return this;
+        }
+
+        private StringFW.Builder field2()
+        {
+            int newBit = fieldsMask | (1 << FIELD_INDEX_FIELD2);
+            assert newBit == (newBit & 0x07) : "Value out of order for field \"field2\" in the list";
+            return field2RW.wrap(buffer(), limit(), maxLimit());
+        }
+
+        public Builder field2(
+            String value)
+        {
```

Comment:
`assert (fieldsMask & ~0x03) == 0;`
where `0x03` means `FIELD_INDEX_FIELD0 | FIELD_INDEX_FIELD1`

if `field0` is required, then
`assert (fieldsMask & 0x01) != 0`
where `0x01` means `FIELD_INDEX_FIELD0`